tests/test_api_typing.py

```diff
@@ -0,0 +1,55 @@
+"""Test module for classes in pandas.api.typing."""
+
+from typing import Literal  # noqa: F401
+from typing import TYPE_CHECKING
+
+import numpy as np
+import pandas as pd
+from pandas.api.typing import (
+    DataFrameGroupBy,
+    DatetimeIndexResamplerGroupby,
+    NAType,
+    SeriesGroupBy,
+)
+from typing_extensions import assert_type
+
+from pandas._typing import Scalar  # noqa: F401
+
+from tests import check
+
+if TYPE_CHECKING:
+    from pandas.core.groupby.groupby import _ResamplerGroupBy  # noqa: F401
+
+
+def test_dataframegroupby():
+    df = pd.DataFrame({"a": [1, 2, 3]})
+    check(
+        assert_type(df.groupby("a"), "DataFrameGroupBy[Scalar, Literal[True]]"),
```

Comment:
We want to make sure that the type the stubs returns is `DataFrameGroupBy`, so use that, not the generic version.